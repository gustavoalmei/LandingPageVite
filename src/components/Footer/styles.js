import { styled, css } from 'styled-components';
import { Container as TextComponet } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../../components/SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${TextComponet} {
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
