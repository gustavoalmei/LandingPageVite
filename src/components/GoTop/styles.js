import { styled, css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
  `}
`;
