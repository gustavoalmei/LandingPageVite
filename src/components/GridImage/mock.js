export const mock = {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    [
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573996/photo_1730755803563_284a0bb30af2_056b06de14.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/eecc7d26262f6a1a80b325d916019656_85279db6cf.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/92eaac8133eb7bd2498bf4a7ab32fc13_b0c64273ab.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/8aed075b2259a6f2bace5c4924ceb0a3_cd7e3d5b77.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/92313bc1e9f73846662edc5681ad4408_f2159488ca.jpg',
        altText: '',
      },
      {
        srcImg:
          'https://res.cloudinary.com/dk0apsprq/image/upload/v1737573997/pexels_souvenirpixels_417074_9e0d95be44.jpg',
        altText: '',
      },
    ],
  ],
};
