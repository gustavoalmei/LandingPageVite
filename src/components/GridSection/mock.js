export const mock = {
  background: false,
  title: 'my grid',
  description:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
  grid: [
    {
      title: 'Teste 1',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    },
    {
      title: 'Teste 2',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    },
    {
      title: 'Teste 3',
      description:
        "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    },
  ],
};
