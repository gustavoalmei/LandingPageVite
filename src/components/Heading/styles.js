import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1.withConfig({
  // withConfig permite configurar como o Styled Components manipula props específicas.
  shouldForwardProp: (prop) => !['colorDark', 'uppercase'].includes(prop),
  // shouldForwardProp é uma função que decide quais props serão passadas para o DOM. Nesse caso, estamos dizendo que colorDark não deve ser repassado.
})`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
