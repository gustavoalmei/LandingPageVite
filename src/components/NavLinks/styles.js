import { styled, css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media (max-width: 768px) {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
