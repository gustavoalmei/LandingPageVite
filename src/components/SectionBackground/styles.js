import { styled, css } from 'styled-components';

const setBackgroundColor = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div.withConfig({
  shouldForwardProp: (prop) => !['background'].includes(prop),
})`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && setBackgroundColor(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
