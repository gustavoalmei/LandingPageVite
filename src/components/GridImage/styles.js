import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    object-fit: cover;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.3) rotate(10deg);
    }
  `}
`;
