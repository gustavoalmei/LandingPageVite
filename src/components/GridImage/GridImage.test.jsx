import { GridImage } from '.';
import { RenderTheme } from '../../styles/handleTheme';

import { mock } from './mock';

describe('<GridImage />', () => {
  it('should render with background', () => {
    const { container } = RenderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without background', () => {
    const { container } = RenderTheme(
      <GridImage {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
