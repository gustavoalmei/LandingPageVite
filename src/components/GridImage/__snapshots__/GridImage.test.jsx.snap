// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should render with background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.5rem;
}

.c2 >.c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  transform: scale(1.3) rotate(10deg);
}

@media (max-width: 768px) {
  .c3 {
    font-size: 3.2rem;
  }
}

@media (max-width: 768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <div
          class="c4 c5"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              class="c8"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage /> should render without background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.5rem;
}

.c2 >.c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  transform: scale(1.3) rotate(10deg);
}

@media (max-width: 768px) {
  .c3 {
    font-size: 3.2rem;
  }
}

@media (max-width: 768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My grid
        </h2>
        <div
          class="c4 c5"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              class="c8"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
